src/main/redux/sagas/index.ts

```diff
@@ -119,6 +130,105 @@ export function* rootSaga() {
+    // Correct HTTP header content-type, but reliance on GitHack servers:
+    // const JSON_URL = `https://raw.githack.com/edrlab/thorium-reader/${BRANCH}/latest.json`;
+    try {
+        let version = IS_DEV ? yield* select((state: RootState) => state.version) : null;
+        // src/main/redux/reducers/index.ts
+        // version: (state: RootState, action: ActionWithSender) => action.type === appActions.initSuccess.ID ? _APP_VERSION : (state?.version ? state.version : null),
+        if (IS_DEV && _APP_VERSION !== version) {
+            debug("VERSION MISMATCH (checkAppVersionUpdate): ", _APP_VERSION, " !== ", version);
+        }
+        if (!version) {
+            version = _APP_VERSION;
+        }
+
+        // yield* call from "typed-redux-saga"
+        // yield call from "redux-saga/effects"
+        const json = yield* callTyped(async (url: string) => {
+
+            // const headers = new Headers();
+            // headers.append("user-agent", "readium-desktop");
+            // headers.append("accept-language", `${locale},en-US;q=0.7,en;q=0.5`);
+
+            const res = await httpGet(url,
+                // { headers },
+            );
+            const ct = parseContentType(res.contentType);
+            if (res.isSuccess) {
+                if (ct === ContentType.Json) {
+                    return await res.response.json();
+                } else if (ct === ContentType.TextPlain) {
+                    try {
+                        return JSON.parse(await res.response.text());
+                    } catch (e) {
+                        debug(e);
+                    }
+                }
+            }
+            return undefined;
+        }, JSON_URL);
+        if (json) {
+            debug(json);
+            try {
+                const date = Date.parse(json.date);
+                debug((new Date(date)).toUTCString());
+                debug((new Date(date)).toString());
+            } catch (e) {
+                debug(e);
+            }
+            if (json.id === "io.github.edrlab.thorium" && json.version && json.url) {
+
+                // Uncomment below to test cases:
+                // json.version = "2.4.0";
+                // json.version = "3.0.0-alpha.1";
+                // json.version = "3.0.0-beta.1";
+                // if (json.version) throw new Error("TESTING...");
+
+                if (semver.gt(json.version, version)) {
+
+                    yield put(versionUpdateActions.notify.build(json.version, json.url));
+
+                    if (IS_DEV) {
+                        yield call(async () => {
+
+                            const translate = diMainGet("translator").translate;
+                            const res = await dialog.showMessageBox(// browserWindow,
+                                {
+                                type: "question",
+                                buttons: [
+                                    translate("app.session.exit.askBox.button.yes"),
```

Comment:
TODO: change i18n key

Reply:
or alternatively, hoist the yes/no generic translated labels to top-level, no need to create "duplicate" entries ;)

Reply:
on est bien d'accord

Reply:
we need a "code mod" script to move this JSON key for all locales